### PR TITLE
Move API side-effects to mutation/query declaration

### LINKED ISSUE #415: Move API side-effects to mutation/query declaration
### Description

Currently, when we want to execute a side-effect (such as a `pushSnackbar`) on the success or failure of a graphql action, we use a `React.useEffect` like so:

```tsx
React.useEffect(() => {
  if (data) {
    pushSnackbar('Success');
  }
},[data]);

// or 

React.useEffect(() => {
  if (error) {
    pushSnackbar('Errorr');
  }
}, [error]);
```
We should map that to the graphql hook itself, using the `onError` and `onCompleted` options of a `useQuery` and `useMutation` hooks

### Acceptance Criteria

- Migrate success side-effects to `onCompleted`
- Migrate failure side-effects to `onError`
- Never use a `React.useEffect` for side-effects that have to do with a query or mutation itself
- Test that the behavior remains the same after the migration

### PR DESCRIPTION
**Pull request by alexmylonas**
## Background

We are using side-effects for the results of queries and mutations, mostly to display a small pop up (snackbar) that displays the result successful or not. 

I moved those to the `onComplete` and `onError` properties of queries/mutations to provide more clarity and consistency.

Closes #415 

## Changes

Removed side-effects from :
- `RemediationButton`, `AnalyticsConsentModal`, `usePolicySuppression`, `UpdateDestinationSidesheet`, `GeneralSettings`, `PromptController`, `PolicyBulkUpload`, `AddDestinationSidesheet`, `ResetUserPassword`

- Refactored `ConfirmModal`

## Testing

- Tested all changes in functionality except `PolicyBulkUpload` locally

### CODE REVIEW

**Comment by alexmylonas** [comment]
@3nvi if you can have a look if I left and unnecessary `React.useEffect` in

**Comment by 3nvi** [comment]
sure,  will  review soon :)

File: web/src/components/sidesheets/AddDestinationSidesheet/AddDestinationSidesheet.tsx [hunk 1]
```diff
@@ -225,18 +229,6 @@ const AddDestinationSidesheet: React.FC<AddDestinationSidesheetProps> = ({ desti
           </IconButton>
           <Heading size="medium">{capitalize(destinationType)} Configuration</Heading>
         </Flex>
-        {addDestinationError && (
```
Comment (3nvi) [inline web/src/components/sidesheets/AddDestinationSidesheet/AddDestinationSidesheet.tsx:228]: we can leave this error as is

File: web/src/components/modals/AnalyticsConsentModal/AnalyticsConsentModal.tsx [hunk 2]
```diff
@@ -48,23 +55,15 @@ const AnalyticsConsentModal: React.FC = () => {
           Opt-in to occasionally provide diagnostic information for improving reliability.
           <b> All information is anonymized.</b>
         </Text>
-        {error ? (
```
Comment (3nvi) [inline web/src/components/modals/AnalyticsConsentModal/AnalyticsConsentModal.tsx:51]: we can leave this error as is

File: web/src/components/sidesheets/PolicyBulkUploadSidesheet/PolicyBulkUploadSidesheet.tsx [hunk 3]
```diff
@@ -127,17 +133,6 @@ const PolicyBulkUploadSideSheet: React.FC<PolicyBulkUploadSideSheetProps> = ({ t
           hidden
           onChange={handleFileChange}
         />
-        {error && (
```
Comment (3nvi) [inline web/src/components/sidesheets/PolicyBulkUploadSidesheet/PolicyBulkUploadSidesheet.tsx:130]: again this error can b epresent here

**Comment by alexmylonas** [comment]
Thanks @3nvi!  It wasn't clear to me what was the correct way there. I have resolved the issues so check and let me know!

File: web/src/components/sidesheets/UpdateDestinationSidesheet/UpdateDestinationSidesheet.tsx [hunk 4]
```diff
@@ -212,18 +216,6 @@ export const UpdateDestinationSidesheet: React.FC<UpdateDestinationSidesheetProp
         <Heading size="medium" mb={8}>
           Update {destination.outputType}
         </Heading>
-        {updateDestinationError && (
```
[the remark this answers is not in the document]
Comment (alexmylonas) [reply web/src/components/sidesheets/UpdateDestinationSidesheet/UpdateDestinationSidesheet.tsx]: Ops sorry slipped through the cracks, i am reverting now!

**Pull request merged**